components/movementsensor/gpsnmea/component.go

```diff
@@ -90,138 +77,17 @@ func (g *NMEAMovementSensor) Start(ctx context.Context) error {
-) (*movementsensor.Accuracy, error) {
-	g.mu.RLock()
-	defer g.mu.RUnlock()
-	acc := movementsensor.Accuracy{
-		AccuracyMap:        map[string]float32{"hDOP": float32(g.data.HDOP), "vDOP": float32(g.data.VDOP)},
-		Hdop:               float32(g.data.HDOP),
-		Vdop:               float32(g.data.VDOP),
-		NmeaFix:            int32(g.data.FixQuality),
-		CompassDegreeError: float32(math.NaN()),
-	}
-	return &acc, g.err.Get()
-}
-
-// LinearVelocity returns the sensor's linear velocity. It requires having a compass heading, so we
-// know which direction our speed is in. We assume all of this speed is horizontal, and not in
-// gaining/losing altitude.
-func (g *NMEAMovementSensor) LinearVelocity(
-	ctx context.Context, extra map[string]interface{},
-) (r3.Vector, error) {
-	g.mu.RLock()
-	defer g.mu.RUnlock()
-
-	if math.IsNaN(g.data.CompassHeading) {
-		return r3.Vector{}, g.err.Get()
-	}
-
-	headingInRadians := g.data.CompassHeading * (math.Pi / 180)
-	xVelocity := g.data.Speed * math.Sin(headingInRadians)
-	yVelocity := g.data.Speed * math.Cos(headingInRadians)
-
-	return r3.Vector{X: xVelocity, Y: yVelocity, Z: 0}, g.err.Get()
-}
-
-// LinearAcceleration returns the sensor's linear acceleration.
-func (g *NMEAMovementSensor) LinearAcceleration(
-	ctx context.Context, extra map[string]interface{},
-) (r3.Vector, error) {
-	return r3.Vector{}, movementsensor.ErrMethodUnimplementedLinearAcceleration
-}
-
-// AngularVelocity returns the sensor's angular velocity.
-func (g *NMEAMovementSensor) AngularVelocity(
-	ctx context.Context, extra map[string]interface{},
-) (spatialmath.AngularVelocity, error) {
-	return spatialmath.AngularVelocity{}, movementsensor.ErrMethodUnimplementedAngularVelocity
-}
-
-// Orientation returns the sensor's orientation.
-func (g *NMEAMovementSensor) Orientation(
-	ctx context.Context, extra map[string]interface{},
-) (spatialmath.Orientation, error) {
-	return nil, movementsensor.ErrMethodUnimplementedOrientation
-}
-
-// CompassHeading returns the heading, from the range 0->360.
-func (g *NMEAMovementSensor) CompassHeading(
-	ctx context.Context, extra map[string]interface{},
-) (float64, error) {
-	g.mu.RLock()
-	defer g.mu.RUnlock()
-
-	lastHeading := g.lastCompassHeading.GetLastCompassHeading()
-	currentHeading := g.data.CompassHeading
-
-	if !math.IsNaN(lastHeading) && math.IsNaN(currentHeading) {
-		return lastHeading, nil
-	}
-
-	if !math.IsNaN(currentHeading) && currentHeading != lastHeading {
-		g.lastCompassHeading.SetLastCompassHeading(currentHeading)
-	}
-
-	return currentHeading, nil
-}
-
-// ReadFix returns Fix quality of MovementSensor measurements.
-func (g *NMEAMovementSensor) ReadFix(ctx context.Context) (int, error) {
-	g.mu.RLock()
-	defer g.mu.RUnlock()
-	return g.data.FixQuality, nil
-}
-
-// ReadSatsInView returns the number of satellites in view.
-func (g *NMEAMovementSensor) ReadSatsInView(ctx context.Context) (int, error) {
-	g.mu.RLock()
-	defer g.mu.RUnlock()
-	return g.data.SatsInView, nil
```

Comment:
We should keep these functions here to be consistent. It doesn't make sense for them to be in cachedData.go since we don't keep track of any of them aside from position and compass heading.